packages/istanbul-lib-source-maps/lib/transformer.js

```diff
@@ -74,12 +84,16 @@ class SourceMapTransformer {
                 }
             });
 
+            const locMapping = branchMeta.loc
+                ? this.resolveMapping(sourceMap, branchMeta.loc, fc.path)
+                : null;
+
             if (!skip && locs.length > 0) {
                 changes += 1;
                 const mappedCoverage = coverageMapper(source);
                 mappedCoverage.addBranch(
                     branchMeta.type,
-                    locs[0] /* XXX */,
+                    locMapping ? locMapping.loc : locs[0],
```

Comment:
Sorry for the delay, I was on vacation. The `resolveMapping` was added just for testability reasons. I did want to create a proper unit test for the transformer instead of an integration test that also used the `source-map` lib. Thus, I mocked the lookup process in transformer.test.js and injected my mock through the constructor. In real-live scenarios the implementation always uses `getMapping` from `get-mapping.js` and the actual `source-map` library.